### ISSUE TITLE
Refactor preview: list rows should have role checkbox

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#88553]
Each list row should have `role = checkbox` so the screen reader would proparly announce to the user that this item is actually a checkbox and that pressing space toggles it.
List supports it. Here's how breakpoints is doing this for example https://github.com/Microsoft/vscode/blob/0b134f8c13b4e19574aaa97a37802a32d725ff0a/src/vs/workbench/contrib/debug/browser/breakpointsView.ts#L94

**Comment by jrieken** [comment]
The [checkboxes](https://github.com/microsoft/vscode/blob/561f5215d73a99203a4e0bd6ea7e856ff869984b/src/vs/workbench/contrib/bulkEdit/browser/bulkEditTree.ts#L231) have that role... @isidorn are you saying the whole row should have it instead or in addition?

**Comment by isidorn** [comment]
Yes the whole row should have it in addition.

**Comment by jrieken** [comment]
I have added the aria provider as suggested by @isidorn but it doesn't seem to be honour by the tree. Is that only for list or not implemented for trees?

**Comment by isidorn** [comment]
It is respected in the listView [here](https://github.com/Microsoft/vscode/blob/160a1c69787f8c97a5edca6ea2ba34bb36710838/src/vs/base/browser/ui/list/listView.ts#L552). 
The list is internally used by the tree, however that options was not being passed. For that I have pushed https://github.com/microsoft/vscode/commit/c96aece131c24c3b3facfcdf042281617f8e1373
@jrieken can you please try it out now? Thanks

fyi @joaomoreno

**Comment by jrieken** [comment]
Didn't help @isidorn... My aria provider still isn't called.

**Comment by isidorn** [comment]
Ok, let me debug it a bit. Thanks for trying out

**Comment by isidorn** [comment]
Forgot one layer. Now it works for me, thus closing.
@jrieken please try it out
@joaomoreno you can review the commit when you find time. Thanks
